src/picongpu/include/plugins/radiation/ExecuteParticleFilter.hpp

```diff
@@ -0,0 +1,92 @@
+/* Copyright 2017 Rene Widera
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "simulation_defines.hpp"
+
+#include "traits/HasIdentifier.hpp"
+
+#include <memory>
+
+
+namespace picongpu
+{
+namespace radiation
+{
+
+    /** read the `radiationFlag` of a species */
+    template< bool hasFilter >
+    struct ExecuteParticleFilter
+    {
+        /** get the attribute value of `radiationFlag`
+         *
+         * @param particle particle to be used
+         * @return value of the attribute `radiationFlag`
+         */
+        template< typename T_Species >
+        void operator()( std::shared_ptr<T_Species> species, const uint32_t currentStep )
+        {
+            auto radiationEnableFunctor = picongpu::radiation::FilterEnableRadiation( currentStep );
+            species->manipulateAllParticles(
+                currentStep,
```

Comment:
it is allowed to add a constructor to the filter functor. 
```
Filter(const unit32_t currentStep);
```
This allows to write time dependent filter.